youtube_dl/extractor/senateisvp.py

```diff
@@ -5,149 +5,169 @@
         'info_dict': {
-            'id': 'commerce011514',
+            'id': 'intel090613',
             'ext': 'mp4',
-            'title': 'Integrated Senate Video Player'
+            'title': 'intel090613',
         },
-        'params': {
-            # m3u8 download
-            'skip_download': True,
+        'expected_warnings': ['Failed to download m3u8 information: HTTP Error 404: Not Found'],
+    }, {
+        'url': 'https://www.senate.gov/isvp/?comm=govtaff&type=archv&stt=975&filename=govtaff111722&auto_play=false&poster=https%3A%2F%2Fwww%2Ehsgac%2Esenate%2Egov%2Fimages%2Fvideo%2Dposter%2Dflash%2Dfit%2Epng',
+        'info_dict': {
+            'id': 'govtaff111722',
+            'ext': 'mp4',
+            'title': 'govtaff111722',
+            'thumbnail': 'https://www.hsgac.senate.gov/images/video-poster-flash-fit.png',
         },
     }, {
-        'url': 'http://www.senate.gov/isvp/?type=arch&comm=intel&filename=intel090613&hc_location=ufi',
-        # checksum differs each time
+        'url': 'https://www.senate.gov/isvp/?type=arch&comm=energy&filename=energy111722&stt=00:22:30&auto_play=false&wmode=transparent&poster=https%3A%2F%2Fwww%2Eenergy%2Esenate%2Egov%2Fthemes%2Fenergy%2Fimages%2Fvideo%2Dposter%2Dflash%2Dfit%2Epng',
         'info_dict': {
-            'id': 'intel090613',
+            'id': 'energy111722',
             'ext': 'mp4',
-            'title': 'Integrated Senate Video Player'
-        }
+            'title': 'energy111722',
+            'thumbnail': 'https://www.energy.senate.gov/themes/energy/images/video-poster-flash-fit.png',
+        },
     }, {
-        # From http://www.c-span.org/video/?96791-1
-        'url': 'http://www.senate.gov/isvp?type=live&comm=banking&filename=banking012715',
-        'only_matching': True,
+        'url': 'https://www.senate.gov/isvp/?comm=foreign&type=archv&stt=0&filename=foreign080322&auto_play=false&wmode=transparent&poster=https%3A%2F%2Fwww%2Eforeign%2Esenate%2Egov%2Fthemes%2Fforeign%2Fimages%2Fvideo%2Dposter%2Dflash%2Dfit%2Epng',
+        'info_dict': {
+            'id': 'foreign080322',
+            'ext': 'mp4',
+            'title': 'foreign080322',
+            'thumbnail': 'https://www.foreign.senate.gov/themes/foreign/images/video-poster-flash-fit.png',
+        },
     }]
 
     @staticmethod
+    # returns url from an iframe
     def _search_iframe_url(webpage):
         mobj = re.search(
-            r"<iframe[^>]+src=['\"](?P<url>https?://www\.senate\.gov/isvp/?\?[^'\"]+)['\"]",
+            r'''<iframe\b[^>]+\bsrc\s*=\s*(['"])(?P<url>https?://www\.senate\.gov/isvp/?\?(?:(?!\1)\S)+)''',
             webpage)
         if mobj:
             return mobj.group('url')
 
+    # returns stream_number, stream_domain, stream_id, and msl3
     def _get_info_for_comm(self, committee):
-        for entry in self._COMM_MAP:
-            if entry[0] == committee:
-                return entry[1:]
+        return self._COMM_MAP[committee][0:]
```

Comment:
If using the original format:
```suggestion
        for entry in self._COMM_MAP:
            if entry[0] == committee:
                return (entry + ['', ''])[1:5]